src/java/org/apache/cassandra/schema/TableMetadata.java

```diff
@@ -461,57 +461,109 @@ public static boolean isSuperColumnMapColumnName(ByteBuffer columnName)
         return !columnName.hasRemaining();
     }
 
+    public void validateTableStructureCompatibility(TableMetadata previous)
+    {
+        if (isIndex())
+            return;
+
+        validateTableName(previous);
+        validateTableType(previous);
+        validatePartitionKeyColumnTypes(previous);
```

Comment:
How is this different than `validatePartitionKeyColumn`? Was there a specific reason to differentiate the checks between this method and `validateCompatibility`?

Reply:
The serialization header does not contain the definition of the individual partition key columns, it only has the partition key type (which is a composite type if the partition key contains multiple columns). For this reason we cannot do the same comparison that would be done between two TableMetadata objects coming from CQL tables - we can only compare the types, and this comparison is sufficient even in case of composite types.

I renamed the method from `validatePartitionKeyColumnTypes` to `validatePartitionKeyTypes` and added a comment to clarify.